### DISCUSSION TITLE
WSL2 and USBIP ,  usb network card not work

### DISCUSSION THREAD

**Discussion by simontg** [microsoft/WSL#6834] [Q&A]
I install on WSL2 machine ( UBUNTU) with this instruction : 
https://github.com/rpasek/usbip-wsl2-instructions
I have old usb network card  ALFA AWUS036H ( Realtek RTL8187).

Before i try connect trough USBIP
In Control Panel and Device Manager 

![image](https://user-images.githubusercontent.com/3414411/115231574-04fd6580-a116-11eb-8da4-d13590263897.png)

When I run usbip on windows , list devices connected to my computer :
![image](https://user-images.githubusercontent.com/3414411/115231619-10509100-a116-11eb-8a8f-ff41c127f2d6.png)

Next step I bind network card
![image](https://user-images.githubusercontent.com/3414411/115231655-17779f00-a116-11eb-8fd1-b4f2a8417be1.png)

And next run –-debug 
On Ubuntu run  ./startusb.sh  , and after that i run command: 
![image](https://user-images.githubusercontent.com/3414411/115231694-23fbf780-a116-11eb-980a-f3d627969091.png)

In the same time on windows show mi something like this :
![image](https://user-images.githubusercontent.com/3414411/115231741-35450400-a116-11eb-8188-9f3f6420cc47.png)

After that i run on Ubuntu lsusb command
![image](https://user-images.githubusercontent.com/3414411/115231767-3e35d580-a116-11eb-9705-459abc9f6bff.png)

But when I run ifconfig , or iwconfig not show anything. 
![image](https://user-images.githubusercontent.com/3414411/115231794-4726a700-a116-11eb-9224-d9a3151ad7f6.png)

I erlier install backports from https://github.com/matrix/backports-rtl8187

I compare WSL2 machine with wmaware machine where this card work normally
![image](https://user-images.githubusercontent.com/3414411/115231833-53aaff80-a116-11eb-8fb1-b4fc4fca117b.png)

When I conected this usb card in linux system I see Driver=rtl8187 .
I think  this is problem. Under WSL2 this card is not correctly regonized.
Somebody try connect usb network card in WSL2 ?
Can anyone chceck how is in his system ?

**Comment by craigloewen-msft** [comment]
I've moved this issue to a discussion. Currently USB over IP is not supported directly in WSL, but perhaps a community member can help you with your questions!

**Reply by goodfore** [reply]
It's most likely the Kernel for WSL2 doesn't have the module for USB NIC enabled. Try to re-complile the kernel with the module and see if it works. I did similar to get the USB mass storage working, following this guide:

https://kumekay.com/compiling-custom-kernel-for-wsl2/

**Comment by gobniqk** [comment]
Experiencing the same problem, trying to find a solution. Will come back if I find one.

**Reply by woadey** [reply]
Any update on this for Win11?

**Reply by gobniqk** [reply]
Not that I found one, preferably I guess it's better to use a vm for this situation. I believe it is possible some way to get it to work, but it would probably be harder and more exhausting than just using another vm. Sad, as WSL2 is very easy to download and use.
